xc/device.ml

```diff
@@ -1489,7 +1489,16 @@ module Dm = struct
 
     let qmp_event_handle domid qmp_event =
       (* This function will be extended to handle qmp events *)
-      debug "Got QMP event, domain-%d: %s" domid qmp_event.event
+      debug "Got QMP event, domain-%d: %s" domid qmp_event.event;
+      qmp_event.data >>= function
+      | RTC_CHANGE timeoffset ->
+        with_xs (fun xs ->
+          let timeoffset_key = sprintf "/vm/%s/rtc/timeoffset" (Uuidm.to_string (Xenops_helpers.uuid_of_domid ~xs domid)) in
+          try
+            let rtc = xs.Xs.read timeoffset_key in
+            xs.Xs.write timeoffset_key Int64.(add timeoffset (of_string rtc) |> to_string)
+          with e -> debug "Failed to process RTC_CHANGE: %s" (Printexc.to_string e)
```

Comment:
The error message should include the domain for which this happened - otherwise hard to debug.

Reply:
Added, thanks.

Reply:
Hi @lindig , I've updated the PR, are you happy to review again?